js/src/forum/addVoteButtons.js

```diff
@@ -28,13 +28,18 @@ export default function () {
 
     extend(CommentPost.prototype, 'actionItems', function (items) {
         const post = this.attrs.post;
+        const user = app.session.user;
         const hasDownvoted = post.hasDownvoted();
         const hasUpvoted = post.hasUpvoted();
 
         const icon = setting('iconName') || 'thumbs';
 
         // We set canVote to true for guest users so that they can access the login by clicking the button
-        const canVote = !app.session.user || post.canVote();
+        let canVote = !user || post.canVote();
+
+        if (user && user.id() === post.user().id() && !setting('allowSelfVote')) {
```

Comment:
This could error if `post.user()` is undefined, right? (e.g. deleted user)

Reply:
Yes, you're right. I didn't know that syntax in JS. I believe it is equal to the PHP 8 one

Reply:
Yes, you're right. I didn't know that syntax in JS. I believe it is equal to the PHP 8 one.

I've added the `?`